pkg/admission/workspacetypeexists/admission.go

```diff
@@ -487,11 +494,15 @@ func validateAllowedParents(parentAliases, childAliases []*tenancyv1alpha1.Works
 func validateAllowedChildren(parentAliases, childAliases []*tenancyv1alpha1.WorkspaceType, parentType, childType logicalcluster.Path) error {
 	var errs []error
 	for _, parentAlias := range parentAliases {
-		if parentAlias.Spec.LimitAllowedChildren == nil || len(parentAlias.Spec.LimitAllowedChildren.Types) == 0 {
+		if parentAlias.Spec.LimitAllowedChildren == nil {
```

Comment:
@sttts While working on the above change, I noticed here that we were skipping the `LimitAllowedChildren.None` check if `Types` was empty, although I would expect Types always be empty if None is set to true? Is that not the case?

If the above assumption is correct, is there any place I could add validation that when `.None` is true, `.Types` should be empty?

Reply:
Might be able to express that using the kube-builder annotations or yaml-patch on the CRD

Reply:
yes, that seems to be a bug.

Reply:
ptal, added CEL validation for the types